src/spikeinterface/qualitymetrics/misc_metrics.py

```diff
@@ -563,7 +563,129 @@ def compute_synchrony_metrics(waveform_extractor, synchrony_sizes=(2, 4, 8), **k
+            firing_rate_histograms[unit_id] = np.concatenate((firing_rate_histograms[unit_id], firing_rates))
+
+    # finally we compute the percentiles
+    firing_ranges = {}
+    for unit_id in unit_ids:
+        firing_ranges[unit_id] = np.percentile(firing_rate_histograms[unit_id], percentiles[1]) - np.percentile(
+            firing_rate_histograms[unit_id], percentiles[0]
+        )
+
+    return firing_ranges
+
+
+_default_params["firing_range"] = dict(bin_size_s=5, percentiles=(0.05, 0.95))
+
+
+def compute_amplitude_spreads(
+    waveform_extractor, num_spikes_per_bin=100, amplitude_extension="spike_amplitudes", unit_ids=None
+):
+    """Calculate spread of spike amplitudes within defined bins of spike events.
+    The spread is the median relative variance (variance divided by the overall amplitude mean)
+    computed over bins of `num_spikes_per_bin` spikes.
+
+    Parameters
+    ----------
+    waveform_extractor : WaveformExtractor
+        The waveform extractor object.
+    num_spikes_per_bin : int, default: 50
+        The number of spikes per bin.
+    amplitude_extension : str, default: 'spike_amplitudes'
+        The name of the extension to load the amplitudes from. 'spike_amplitudes' or 'amplitude_scalings'.
+    unit_ids : list or None
+        List of unit ids to compute the amplitude spread. If None, all units are used.
+
+    Returns
+    -------
+    amplitude_spreads : dict
+        The amplitude spread for each unit.
+
+    Notes
+    -----
+    Designed by Simon Musall and ported to SpikeInterface by Alessio Buccino.
+    """
+    assert amplitude_extension in (
+        "spike_amplitudes",
+        "amplitude_scalings",
+    ), "Invalid amplitude_extension. It can be either 'spike_amplitudes' or 'amplitude_scalings'"
+    sorting = waveform_extractor.sorting
+    spikes = sorting.to_spike_vector()
+    num_spikes = sorting.count_num_spikes_per_unit()
+    if unit_ids is None:
+        unit_ids = sorting.unit_ids
+
+    if waveform_extractor.is_extension(amplitude_extension):
+        sac = waveform_extractor.load_extension(amplitude_extension)
+        amps = sac.get_data(outputs="concatenated")
+        if amplitude_extension == "spike_amplitudes":
+            amps = np.concatenate(amps)
+    else:
+        warnings.warn("")
+        empty_dict = {unit_id: np.nan for unit_id in unit_ids}
+        return empty_dict
+
+    all_unit_ids = list(sorting.unit_ids)
+    amplitude_spreads = {}
+    for unit_id in unit_ids:
+        amps_unit = amps[spikes["unit_index"] == all_unit_ids.index(unit_id)]
+        amp_mean = np.abs(np.mean(amps_unit))
+        if num_spikes[unit_id] < num_spikes_per_bin:
+            amp_spread = np.std(amps_unit) / amp_mean
+        else:
+            amp_spreads = []
+            for i in range(0, num_spikes[unit_id], num_spikes_per_bin):
+                amp_spreads.append(np.std(amps_unit[i : i + num_spikes_per_bin]) / amp_mean)
+            amp_spread = np.median(amp_spreads)
```

Comment:
```suggestion
            amp_spread = np.percentile(amplitude_spreads, percentiles[1]) - np.percentile(amplitude_spreads, percentiles[0])
```

I think that using the median here is not doing what we want. If I understand correctly, this metric should spot/quantify amplitudes like the following:

![image](https://github.com/SpikeInterface/spikeinterface/assets/17097257/fac27268-c343-470e-8da0-8e30d7118cae)

where probably the "bursts" of high-amplitude spiking activity are noise events. If the number of bins with normal activity is greater than the ones with high noisy activity, this behavior will not be captured by the median. Mean would be better, or even percentile distance (like the firing range). What do you think?

Reply:
I think that makes total sense and as we discussed that was the original idea of the metric. The way it is currently implemented with the median would return the variance of the amplitudes but ignore these bursts that are probably triggered by noise events. It seems to me like both of these (median and percentile distance) are useful but describe different cluster features. Would it be possible to return both or would these be two different metrics then?